### ISSUE TITLE
Long term caching support

### ISSUE THREAD

**Issue by kamsar** [connectivedx/fuzzy-chainsaw#177]
## Expected Behavior
Since images and fonts already get hashes in their filenames for cache busting it'd be nice if CSS and JS got the same treatment. As an _option_, for systems (like Sitecore) that can easily read hashed asset names.

## Current Behavior
Enabling long caching for performance results in faux bug reports due to stale caches.

## Possible Solution

Here is some prototype code that does this for one project. It **does not** fix references in the static site generator, so when this code is active (in the `webpack.production.config.js` at the bottom) it breaks the styleguide.

```js
  // set output filename hashing on for all js output files
  if(config.output.filename) {
    config.output.filename = hashify(config.output.filename);
  }

  // set output filename hashing on for all css output files (via ExtractTextPlugin)
  config.plugins
    .filter(plugin => plugin.filename)
    .forEach(plugin => plugin.filename = hashify(plugin.filename));

  return config;
});

// takes a path without a hash and adds one, e.g.
// /foo/bar.js -> /foo/bar-[hash].js
// (webpack adds the actual hash value)
function hashify(existingPath) {
  const extensionIndex = existingPath.lastIndexOf('.');
  const extension = existingPath.substring(extensionIndex);
  const fileName = existingPath.substring(0, extensionIndex);

  return `${fileName}-[hash]${extension}`;
}
```

## Issues

As previously mentioned the JSX generated HTML is unaware of the hashed asset names and fails. That would need to be resolved before using this code. You can [get hashed filenames](https://webpack.github.io/docs/long-term-caching.html) out of webpack though.

**Comment by krambuhl** [comment]
webpack provides some long term caching tools.  We should be abled to update this section: https://github.com/connectivedx/fuzzy-chainsaw/blob/develop/build/webpack/webpack.factory.js#L62

```
outputScript: '/assets/[name]-[hash:8].js',
outputStyle: '/assets/[name]-[hash:8].css'
```

and get hashed files.

https://webpack.github.io/docs/long-term-caching.html describes that process and how we would grab the output filenames from the stats.

**Comment by krambuhl** [comment]
This has been resolved on the `2.0` branch.

Custom naming patterns can now be defined in the `source/fc-config.js` file.
